src/reformat_libyuv.c

```diff
@@ -50,6 +56,126 @@ unsigned int avifLibYUVVersion(void)
 #pragma clang diagnostic pop
 #endif
 
+static avifResult avifImageRGBToYUVLibYUV8bpc(avifImage * image, const avifRGBImage * rgb);
+
+avifResult avifImageRGBToYUVLibYUV(avifImage * image, const avifRGBImage * rgb)
+{
+    if ((image->depth == 8) && (rgb->depth == 8)) {
+        return avifImageRGBToYUVLibYUV8bpc(image, rgb);
+    }
+
+    // This function didn't do anything; use the built-in conversion.
+    return AVIF_RESULT_NOT_IMPLEMENTED;
+}
+
+// Returns whether libyuv was compiled with LIBYUV_BIT_EXACT defined or not.
+static avifBool avifIsLibyuvBitExact()
+{
+    // Use hardcoded conversion results to detect precision loss.
+    // These values may need to be updated if libyuv behavior changes.
+
+    // libyuv ARGB (libavif BGRA) to YUV limited range BT.601
+    const uint8_t argb[] = { 215, 22, 5, 255 };
+    uint8_t y = 0, u = 0, v = 0;
+    ARGBToI444(argb, 0, &y, 0, &u, 0, &v, 0, /*width=*/1, /*height=*/1);
+    if (y == 49 && u == 215 && v == 107) { // leads to BGR 212, 21, 5 when converted back with libyuv
+        return AVIF_TRUE;
+    }
+    assert(y == 49 && u == 214 && v == 106); // leads to BGR 210, 23, 3 when converted back with libyuv
+    return AVIF_FALSE;
+}
+
+avifResult avifImageRGBToYUVLibYUV8bpc(avifImage * image, const avifRGBImage * rgb)
+{
+    assert((image->depth == 8) && (rgb->depth == 8));
+    // libavif uses byte-order when describing pixel formats, such that the R in RGBA is the lowest address,
+    // similar to PNG. libyuv orders in word-order, so libavif's RGBA would be referred to in libyuv as ABGR.
+
+    if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV400) {
+        // Generic mapping from any RGB layout (with or without alpha) to monochrome.
+        int (*RGBtoY)(const uint8_t *, int, uint8_t *, int, int, int) = NULL;
+
+        // LIBYUV_BIT_EXACT has no impact on monochrome conversions.
+        if (image->yuvRange == AVIF_RANGE_LIMITED) {
+            if (rgb->format == AVIF_RGB_FORMAT_BGRA) {
+                RGBtoY = ARGBToI400;
+            }
+        } else { // image->yuvRange == AVIF_RANGE_FULL
+            if (rgb->format == AVIF_RGB_FORMAT_BGRA) {
+                RGBtoY = ARGBToJ400;
+            }
+        }
+
+        return !RGBtoY ? AVIF_RESULT_NOT_IMPLEMENTED
+               : RGBtoY(rgb->pixels,
+                        rgb->rowBytes,
+                        image->yuvPlanes[AVIF_CHAN_Y],
+                        image->yuvRowBytes[AVIF_CHAN_Y],
+                        image->width,
+                        image->height)
+                   ? AVIF_RESULT_REFORMAT_FAILED
+                   : AVIF_RESULT_OK;
```

Comment:
Nit: I think it'd be clearer to rewrite this using if statements:
```
        if (!RGBtoY) {
            return AVIF_RESULT_NOT_IMPLEMENTED;
        }
        if (RGBtoY(rgb->pixels, ...)) {
            AVIF_RESULT_REFORMAT_FAILED;
        }
        return AVIF_RESULT_OK;
```

Please make a similar change to the return statement at line 164.